tests/unit/modules/test_elasticsearch.py

```diff
@@ -2264,3 +2264,115 @@ def delete_template(self, id=None):
         with patch.object(elasticsearch, '_get_instance',
                           MagicMock(return_value=MockElastic())):
             self.assertRaises(CommandExecutionError, elasticsearch.search_template_delete, "foo")
+
+    def test_cluster_get_settings_succeess(self):
+        '''
+        Test if cluster get_settings fetch succeeds
+        '''
+        class MockElasticCluster(object):
+            '''
+            Mock of Elasticsearch ClusterClient
+            '''
+            def get_settings(self, flat_settings=False, include_defaults=False):
+                '''
+                Mock of get_settings method
+                '''
+                return {"transient": {}, "persistent": {}}
+
+        class MockElastic(object):
+            '''
+            Mock of Elasticsearch client
+            '''
+            cluster = MockElasticCluster()
+
+        with patch.object(elasticsearch, '_get_instance',
+                          MagicMock(return_value=MockElastic())):
+            self.assertDictEqual(elasticsearch.cluster_get_settings(), {"transient": {}, "persistent": {}})
```

Comment:
I'm not sure if you understood what I was talking about on Slack, but because there really isn't *behavior* to test here it's probably reasonable to just make sure that interfaces aren't changing without warning. You actually don't need to write your own mock classes - you can just use a (series) of MagicMocks for that

```
expected_settings = {"transient": {}, "persistent": {}}
fake_instance = MagicMock()
fake_instance.cluster = MagicMock()
fake_instance.cluster.get_settings = MagicMock(return_value=expected_settings)
with patch.object(elasticsearch, '_get_instance', fake_instance):
     actual_settings = elasticsearch.cluster_get_settings()
     fake_instance.cluster.get_settings.assert_called_with(flat_settings=None, include_defaults=None)
```

I'm not a *huge* fan of deep mocking like this - ideally you shouldn't be mocking anything - but I'm not aware of any better patterns to follow for cases like this.

If I had the [infinity gauntlet](https://en.wikipedia.org/wiki/The_Infinity_Gauntlet), I would write a functional test here. That is, I would have an Elasticsearch cluster for real that I could test with, that I would spin up at the beginning of the Elasticsearch tests and down when I am finished testing. Then I would test that in fact, settings were set and could also be retrieved.

That might be prohibitive, though, so the pragmatist in me says that just making sure that we're passing through the values that we expect to pass in, and we're returning the values that we expect to return is good enough, for now.

I'm open to other suggestions, though!

Reply:
Well, I'm tried to make changes which are aligned with the rest of the changes already present ie "copy-paste" :-D I do hope they make at least some sense.

Reply:
Yeah, it does make some sense. Do you understand what I meant Re: mocks, though? Does that feel like a better approach than the one that we've got here?

I'd also recommend adding some comment to the effect of, "Yes, we're aware that these tests aren't *super* useful - more of just a smoke test to make sure we don't accidentally break things". Seem reasonable?

Reply:
Well yes, I'm trying to understand your example which is not working for me(yet). I mean i get the idea now I'm trying to understand what is wrong with the syntax(I don't have much experience with this whole mock thingy)

Reply:
tests were updated based on your proposal